sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridPublisherClient.cs

```diff
@@ -307,16 +312,20 @@ private async Task<Response> PublishCustomEventsInternal(IEnumerable<object> eve
         /// <param name="key">Key credential used to generate the token.</param>
         /// <param name="apiVersion">Service version to use when handling requests made with the SAS token.</param>
         /// <returns>Returns the generated SAS token string.</returns>
-        public static string BuildSharedAccessSignature(Uri endpoint, DateTimeOffset expirationUtc, AzureKeyCredential key, string apiVersion = "2018-01-01")
+        public static string BuildSharedAccessSignature(Uri endpoint, DateTimeOffset expirationUtc, AzureKeyCredential key, EventGridPublisherClientOptions.ServiceVersion apiVersion = EventGridPublisherClientOptions.LatestVersion)
         {
             const char Resource = 'r';
             const char Expiration = 'e';
             const char Signature = 's';
 
             var uriBuilder = new RequestUriBuilder();
             uriBuilder.Reset(endpoint);
-            uriBuilder.AppendQuery("api-version", apiVersion, true);
-            string encodedResource = HttpUtility.UrlEncode(endpoint.ToString());
+            if (uriBuilder.Query.Contains("api-version"))
```

Comment:
not sure if this is safe. Maybe we can use HttpUtility.ParseQueryString.
Did we test what happens if the api-version is already in the URI?
I just looked at the JS implementation and they don't do any sort of validation to prevent the version from being added twice, so mayb we just leave this out?
https://github.com/Azure/azure-sdk-for-js/pull/10059/files#diff-c6b9f1c0828da0c4eb413e022e5a862fR34

Reply:
It seems to append the query no matter what like this: ?api-version=2018-01-01&api-version=2018-01-01

Reply:
I mean does the service complain at all?

Reply:
No

Reply:
Ok, I would just leave out any validation around this. We just assume the Uri the user provides does not contain it but if it does it is up to the service to decide what to do.